source/common/http/codec_client.h

```diff
@@ -129,7 +129,7 @@ class CodecClient : protected Logger::Loggable<Logger::Id::client>,
   CodecType type() const { return type_; }
 
   // Note this is the L4 stream info, not L7.
-  const StreamInfo::StreamInfo& streamInfo() { return connection_->streamInfo(); }
```

Comment:
Your PR description says
" Set downstream filter state on upstream stream info."
but you're setting downstream filter state on the upstream _connection_ stream info.

Matching downstream L7 streams with upstream L4 connections seems like it's going to be problematic.  Is that what you're intending to do?  I would think that we'd want to match stream to stream, in case multiple L7 streams went out over one virtual connection.

Reply:
Hmmm, this is a very good point. I think this PR needs more thought actually. L4 connections from downstreams are not strictly one-to-one with L4 upstream connections so it's not right to use either L4 or L7 filter state here. I tend to think that the import of filter state must also be explicit like in https://github.com/envoyproxy/envoy/pull/19435. That way the hash policy determines the grouping of downstream L4/L7 stream filter states.